cpp/example_code/cognito/README.md

```diff
@@ -0,0 +1,86 @@
+# Amazon Cognito code examples for the SDK for C++
+
+## Overview
+
+Shows how to use the AWS SDK for C++ with Amazon Cognito to sign up users, set users up for multi-factor authentication (MFA), and sign in to
+get access tokens.
+
+*Amazon Cognito provides authentication, authorization, and user management for your web and mobile apps. Your users can sign in directly with a user name and password, or through a third party such as Facebook, Amazon, Google or Apple.*
+
+## ⚠️ Important
+
+* Running this code might result in charges to your AWS account. 
+* Running the tests might result in charges to your AWS account.
+* We recommend that you grant your code least privilege. At most, grant only the minimum permissions required to perform the task. For more information, see [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege). 
+* This code is not tested in every AWS Region. For more information, see [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).
+
+## Code examples
+
+### Single actions
+
+Code excerpts that show you how to call individual service functions.
+
+* [Confirm a user](./getting_started_with_user_pools.cpp) (ConfirmSignUp)
+* [Delete a user](./getting_started_with_user_pools.cpp) (DeleteUser)
+* [Get a token to associate an MFA application with a user](./getting_started_with_user_pools.cpp) (AssociateSoftwareToken)
+* [Get information about a user](./getting_started_with_user_pools.cpp) (AdminGetUser)
+* [Resend a confirmation code](./getting_started_with_user_pools.cpp) (ResendConfirmationCode)
+* [Respond to SRP authentication challenges](./getting_started_with_user_pools.cpp) (RespondToAuthChallenge)
+* [Sign up a user](./getting_started_with_user_pools.cpp) (SignUp)
+* [Start authentication with a tracked device](./getting_started_with_user_pools.cpp) (InitiateAuth)
+* [Verify an MFA application with a user](./getting_started_with_user_pools.cpp) (VerifySoftwareToken)
+
```

Comment:
The only thing I see missing (from here, and from the code and metadata) is this part of the spec, should it be included?

> 8. Sign in again, get prompted to submit TOTP: AdminInitiateAuth with ADMIN_USER_PASSWORD_AUTH (Response: “ChallengeName”: “SOFTWARE_TOKEN_MFA”)
> 9. Provide TOTP, get tokens: AdminRespondToAuthChallenge

Reply:
I am going to correct that by adding the optional part.